server/src/main/java/org/elasticsearch/common/collect/ImmutableOpenMap.java

```diff
@@ -163,6 +168,26 @@ public void remove() {
         };
     }
 
+    /**
+     * Returns a sequential unordered stream of the map entries.
+     *
+     * @return a {@link Stream} of the map entries as {@link Entry}
+     */
+    public Stream<Entry<KType, VType>> stream() {
+        Iterator<ObjectObjectCursor<KType, VType>> mapIterator = map.iterator();
+        return StreamSupport.stream(new Spliterators.AbstractSpliterator<>(map.size(), Spliterator.SIZED) {
```

Comment:
I think we can add `DISTINCT` as a characteristic?

Reply:
Good point! Will add it.